src/org/jmc/gui/PreviewPanel.java

```diff
@@ -771,8 +771,8 @@ public void mouseDragged(MouseEvent e) {
 
 		if(moving_map)
 		{		
-			shift_x+=(x-last_x)/zoom_level;
-			shift_y+=(y-last_y)/zoom_level;
+			shift_x+=(float)(x-last_x)/zoom_level;
+			shift_y+=(float)(y-last_y)/zoom_level;

```

Comment:
It looks like it is unneccessary, but remember what my first comment said, we need more precision for the preview panel dragging smooth. In java, if both integers are divided, the resulting value will be integer again so therefore we have to cast either one of the operands, left or right into floats, then the resulting expression will give us floats, it has more precision, making preview panel dragging smoother.

So yes, you have to keep in mind the datatype when you are doing an expression. If you want float result, we need either operands to be float, so cast it to float. Give it try, remove that cast and you will see that as soon as you drag around, you will be teleported to nowhere

Reply:
I tried it without the cast and it worked fine for me, since zoom_level is a float the division op ends up with float values.

Reply:
you are correct, that cast is unneccessary, my bad i thought both operands are integers... fixing that real quick now

Reply:
yea, I know, I can zoom right in and it pans perfectly smoothly even without that cast since the zoom_level variable is a float so java does a float division. We can keep it anyway, doesn't matter that much.

Reply:
okay then, its up to you now to close the pull request